neural_lam/models/ar_model.py

```diff
@@ -58,21 +91,30 @@ def __init__(self, args):
             self.num_grid_nodes,
             grid_static_dim,
         ) = self.grid_static_features.shape
+
         self.grid_dim = (
-            2 * self.config_loader.num_data_vars()
+            2 * self.grid_output_dim
             + grid_static_dim
-            + self.config_loader.dataset.num_forcing_features
+            + num_forcing_vars * forcing_window_size
         )
 
         # Instantiate loss function
         self.loss = metrics.get_metric(args.loss)
 
+        boundary_mask = torch.tensor(
+            da_boundary_mask.values, dtype=torch.float32
+        ).unsqueeze(
+            1
+        )  # add feature dim
+
+        self.register_buffer("boundary_mask", boundary_mask, persistent=False)
         # Pre-compute interior mask for use in loss function
         self.register_buffer(
-            "interior_mask", 1.0 - self.border_mask, persistent=False
+            "interior_mask", 1.0 - self.boundary_mask, persistent=False
         )  # (num_grid_nodes, 1), 1 for non-border
 
-        self.step_length = args.step_length  # Number of hours per pred. step
+        # Number of hours per pred. step
+        self.step_length = datastore.step_length
```

Comment:
As the datastore is an attribute and we can get the step length from there I don't think there is any need to put this as another attribute. If we also get rid of it's use in `plot_error_map` (see other comment there) it is also hardly used here.

Reply:
Done in https://github.com/leifdenby/neural-lam/commit/2bbe666fab3df2be1471b7eaebffa245cef48716